### ISSUE TITLE
Layout View Resorting is breaking Windows Narrators ability to navigate through elements

### ISSUE THREAD

**Issue by PureWeen** [dotnet/maui#6140]
### Description

The LayoutHandler sorts all of its children via `OrderByZIndex` which uses an `Array.Sort` and a comparer. This causes the elements to all change logical position when they added to the `Panel`. `Narrator` is unable to reconstruct the semantic order just by the location of the elements on the screen.

If you remove the `OrderByZIndex` call then `Narrator` works without any issues. `OrderByZIndex` also appears to break TabIndex ordering as well.

### Steps to Reproduce

1. Load up controls sample gallery
2. go to the buttons controls page
3. turn on narrator 
4. hit the tab key (or swipe forward) to navigate through elements

### Possible workarounds

- Fix the `OrderByZIndex` so all elements with a matching `Zindex` maintain the same logical order they are added to the layout as. I do wonder if this solution still doesn't quite work because it will still place two elements logically very far apart even though they visually aren't.
- We're going to have to create a custom automation peer for layouts to resolve some other issues, so in theory we could add a custom automation peer that's able to return the [elements ](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.getchildrencore?view=winrt-22000) in the logical order the user added them in. 

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

 Windows

### Affected platform versions

WinUI

**Comment by PureWeen** [comment]
Fixed by https://github.com/dotnet/maui/pull/7270
